src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/CLong.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+#pragma warning disable SA1121 // We use our own aliases since they differ per platform
+#if TARGET_WINDOWS
+using NativeType = System.Int32;
+#else
+using NativeType = System.IntPtr;
+#endif
+
+namespace System.Runtime.InteropServices
+{
+    [CLSCompliant(false)]
```

Comment:
I think `CLong` can be `CLSCompliant` since it is signed and internally an `int` or `nint`, which both are `CLSCompliant`. However, I don't think `CLSCompliant` means it isn't consumable by VB.NET, using other types like `UIntPtr` or `SByte` seem to work just fine: https://sharplab.io/#v2:DYLgbgRgNALiCWwA+BJAtgBwPYCcYGcACAZQE98YBTNAWACgAFAVwmHgGNCBhYAQ3yJd6hEYWasOhAGJMAduxjwsswgFkAjAAoAlIQCCRAKopZMBjBzDR1gEqUYTHCuOnzOAHQAtSjixWRAKKyACbScgpKsv6E0eJsnDLyispqAEw6+kTEAEKkVNG29o4qALTq0UGhiRHK9JXcfAJAA=

I'm no expert in VB though, so it's definitely plausible that I'm unaware of some detail 😄

Reply:
`CLSCompilant(false)` just means that some or all of the members use types that are not guaranteed to be usable from all languages. The types themselves are still usable, but some members (like the `uint` constructor on `CULong`) aren't supported. It doesn't block the types from being used in VB.NET.

Reply:
Is there anything actually preventing the use of this from other languages given it is just a struct wrapping a `CLSCompliant` type?